{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a5009ed7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import yaml\n",
    "import json\n",
    "import pandas as pd\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0a7f9352",
   "metadata": {},
   "outputs": [],
   "source": [
    "cwl_dir = '../cwl/'\n",
    "Path(cwl_dir).mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "1de0bb82",
   "metadata": {},
   "outputs": [],
   "source": [
    "template = {\n",
    "    'maf': {\n",
    "            'class': 'File',\n",
    "            'path': 'a/file/path'\n",
    "    },\n",
    "    'bam': {\n",
    "            'class': 'File',\n",
    "            'path': 'a/file/path'\n",
    "    },\n",
    "    'ref_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/dir/path'\n",
    "    },\n",
    "    'bed': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'f_allele': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'netmhc': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'input_type': 'a_string' # rna or dna, default is dna\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3716c24d",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(template, open(os.path.join(cwl_dir, 'template.neoscan.yaml'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "e2d4236d",
   "metadata": {},
   "outputs": [],
   "source": [
    "cwl = {\n",
    "    'class': 'CommandLineTool',\n",
    "    'cwlVersion': 'v1.0',\n",
    "    'id': 'neoscan',\n",
    "    'label': 'neoscan',\n",
    "    'baseCommand': ['perl', '/pecgs-neoscan/src/neoscan.py'],\n",
    "    'inputs': [\n",
    "        {\n",
    "            'id': 'maf',\n",
    "            'type': 'File',\n",
    "            'inputBinding': {\n",
    "                'position': '1',\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'id': 'bam',\n",
    "            'type': 'File',\n",
    "            'secondaryFiles': [\n",
    "               '$(self.basename).bai',\n",
    "            ],\n",
    "            'inputBinding': {\n",
    "                'position': '2',\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'id': 'ref_dir',\n",
    "            'type': 'Directory',\n",
    "            'inputBinding': {\n",
    "                'prefix': '--ref-dir',\n",
    "                'position': '0'\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'id': 'bed',\n",
    "            'type': 'File',\n",
    "            'inputBinding': {\n",
    "                'prefix': '--bed',\n",
    "                'position': '0'\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'id': 'f_allele',\n",
    "            'type': 'File',\n",
    "            'inputBinding': {\n",
    "                'prefix': '--f-allele',\n",
    "                'position': '0'\n",
    "            }\n",
    "        },\n",
    "                {\n",
    "            'id': 'netmhc',\n",
    "            'type': 'File',\n",
    "            'inputBinding': {\n",
    "                'prefix': '--netmhc',\n",
    "                'position': '0'\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'id': 'input_type',\n",
    "            'type': 'string?',\n",
    "            'default': 'dna',\n",
    "            'inputBinding': {\n",
    "                'prefix': '--input-type',\n",
    "                'position': '0'\n",
    "            }\n",
    "        },\n",
    "        # needs path to be set so it works on compute1\n",
    "        {\n",
    "            'id': 'environ_PATH',\n",
    "            'type': 'string?',\n",
    "            'default': '/miniconda/envs/neoscan/bin:$PATH'\n",
    "        },\n",
    "    ],\n",
    "    'arguments': [\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--out-dir',\n",
    "            'valueFrom': 'output'\n",
    "        },\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--log-dir',\n",
    "            'valueFrom': 'logs'\n",
    "        },\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--neoscan-dir',\n",
    "            'valueFrom': '/pecgs-neoscan/src/neoscan'\n",
    "        },\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--optitype-script',\n",
    "            'valueFrom': '/miniconda/envs/neoscan/bin/OptiPathPipeline.py'\n",
    "        },\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--f-opti-config',\n",
    "            'valueFrom': '/pecgs-neoscan/src/neoscan/config.ini'\n",
    "        },\n",
    "        \n",
    "    ],\n",
    "    'outputs': [\n",
    "        {\n",
    "            'id': 'snv_summary',\n",
    "            'type': 'File',\n",
    "            'outputBinding': {'glob': 'output/sample.neo.snv.summary'},\n",
    "        },\n",
    "        {\n",
    "            'id': 'indel_summary',\n",
    "            'type': 'File',\n",
    "            'outputBinding': {'glob': 'output/sample.neo.indel.summary'},\n",
    "        },\n",
    "    ],\n",
    "    'requirements': [\n",
    "        {\n",
    "            'class': 'DockerRequirement',\n",
    "            'dockerPull': 'estorrs/pecgs-neoscan:0.0.1'\n",
    "        },\n",
    "        {\n",
    "            'class': 'ResourceRequirement',\n",
    "            'ramMin': 28000,\n",
    "        },\n",
    "        {\n",
    "            'class': 'EnvVarRequirement',\n",
    "            'envDef': {\n",
    "                'PATH': '$(inputs.environ_PATH)',\n",
    "            }\n",
    "        }\n",
    "    ]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "8fc6fa17",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(cwl, open(os.path.join(cwl_dir, 'neoscan.cwl'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e41f2630",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
