{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "ebd0c356",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import yaml\n",
    "import json\n",
    "import pandas as pd\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "b5837237",
   "metadata": {},
   "outputs": [],
   "source": [
    "cwl_dir = '../cwl/'\n",
    "Path(cwl_dir).mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "2d486fc1",
   "metadata": {},
   "outputs": [],
   "source": [
    "template = {\n",
    "    'maf': {\n",
    "            'class': 'File',\n",
    "            'path': 'a/file/path'\n",
    "    },\n",
    "    'bam': {\n",
    "            'class': 'File',\n",
    "            'path': 'a/file/path'\n",
    "    },\n",
    "    'ref_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/dir/path'\n",
    "    },\n",
    "    'bed': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'input_type': 'a_string' # rna or dna, default is dna\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "5e16fed9",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(template, open(os.path.join(cwl_dir, 'template.neoscan.yaml'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "3415b70b",
   "metadata": {},
   "outputs": [],
   "source": [
    "cwl = {\n",
    "    'class': 'CommandLineTool',\n",
    "    'cwlVersion': 'v1.0',\n",
    "    'id': 'neoscan',\n",
    "    'label': 'neoscan',\n",
    "    'baseCommand': ['/usr/bin/python', '/pecgs-neoscan/src/neoscan.py'],\n",
    "    'inputs': [\n",
    "        {\n",
    "            'id': 'maf',\n",
    "            'type': 'File',\n",
    "            'inputBinding': {\n",
    "                'position': '1',\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'id': 'bam',\n",
    "            'type': 'File',\n",
    "            'secondaryFiles': [\n",
    "               '$(self.basename).bai',\n",
    "            ],\n",
    "            'inputBinding': {\n",
    "                'position': '2',\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'id': 'ref_dir',\n",
    "            'type': 'Directory',\n",
    "            'inputBinding': {\n",
    "                'prefix': '--ref-dir',\n",
    "                'position': '0'\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'id': 'bed',\n",
    "            'type': 'File',\n",
    "            'inputBinding': {\n",
    "                'prefix': '--bed',\n",
    "                'position': '0'\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'id': 'input_type',\n",
    "            'type': 'string?',\n",
    "            'default': 'dna',\n",
    "            'inputBinding': {\n",
    "                'prefix': '--input-type',\n",
    "                'position': '0'\n",
    "            }\n",
    "        },\n",
    "        # needs path to be set so it works on compute1\n",
    "        {\n",
    "            'id': 'environ_PATH',\n",
    "            'type': 'string?',\n",
    "#             'default': '/miniconda/envs/neoscan/bin:$PATH'\n",
    "            'default': '/usr/local/bin/OptiType:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/conda/bin:/home/biodocker/bin:/miniconda/envs/neoscan/bin:/miniconda/bin:$PATH' # just hardcoding here is the easiest thing for now\n",
    "        },\n",
    "    ],\n",
    "    'arguments': [\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--out-dir',\n",
    "            'valueFrom': 'output'\n",
    "        },\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--log-dir',\n",
    "            'valueFrom': 'logs'\n",
    "        },\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--neoscan-dir',\n",
    "            'valueFrom': '/pecgs-neoscan/src/neoscan'\n",
    "        },\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--optitype-script',\n",
    "            'valueFrom': '/usr/local/bin/OptiType/OptiTypePipeline.py'\n",
    "        },\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--f-opti-config',\n",
    "            'valueFrom': '/pecgs-neoscan/src/neoscan/config.ini'\n",
    "        },\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--netmhc',\n",
    "            'valueFrom': '/pecgs-neoscan/src/netmhc/netMHC-4.0/netMHC'\n",
    "        },\n",
    "                {\n",
    "            'position': 0,\n",
    "            'prefix': '--f-allele',\n",
    "            'valueFrom': '/pecgs-neoscan/src/netmhc/netMHC-4.0/data/allelelist'\n",
    "        },\n",
    "        \n",
    "    ],\n",
    "    'outputs': [\n",
    "        {\n",
    "            'id': 'snv_summary',\n",
    "            'type': 'File',\n",
    "            'outputBinding': {'glob': 'output/sample.neo.snv.summary'},\n",
    "        },\n",
    "        {\n",
    "            'id': 'indel_summary',\n",
    "            'type': 'File',\n",
    "            'outputBinding': {'glob': 'output/sample.neo.indel.summary'},\n",
    "        },\n",
    "    ],\n",
    "    'requirements': [\n",
    "        {\n",
    "            'class': 'DockerRequirement',\n",
    "            'dockerPull': 'estorrs/pecgs-neoscan:0.0.1'\n",
    "        },\n",
    "        {\n",
    "            'class': 'ResourceRequirement',\n",
    "            'ramMin': 100000,\n",
    "            'coresMin': 4\n",
    "        },\n",
    "        {\n",
    "            'class': 'EnvVarRequirement',\n",
    "            'envDef': {\n",
    "                'PATH': '$(inputs.environ_PATH)',\n",
    "            }\n",
    "        }\n",
    "    ]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "967122b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(cwl, open(os.path.join(cwl_dir, 'neoscan.cwl'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c420a372",
   "metadata": {},
   "outputs": [],
   "source": [
    "export LSF_DOCKER_VOLUMES=\"/storage1/fs1/dinglab/Active/Projects/estorrs/wombat/tests/data/neoscan:/storage1/fs1/dinglab/Active/Projects/estorrs/wombat/tests/data/neoscan /storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline:/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/neoscan/C3L-00677:/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/neoscan/C3L-00677 /storage1/fs1/dinglab/Active:/storage1/fs1/dinglab/Active /storage1/fs1/m.wyczalkowski/Active:/storage1/fs1/m.wyczalkowski/Active /scratch1/fs1/dinglab/estorrs:/scratch1/fs1/dinglab/estorrs\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5cf3bd85",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd /pecgs-neoscan/src/neoscan\n",
    "export PATH=\"/miniconda/envs/neoscan/bin:$PATH\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "96205e8b",
   "metadata": {},
   "outputs": [],
   "source": [
    "perl neoscan.pl --rdir /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/neoscan/C3L-00677/cromwell-workdir/cromwell-executions/neoscan.cwl/6994fb4a-0450-4dbb-818a-bcfdc8d73721/call-neoscan.cwl/execution/output --bed /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/neoscan/C3L-00677/cromwell-workdir/cromwell-executions/neoscan.cwl/6994fb4a-0450-4dbb-818a-bcfdc8d73721/call-neoscan.cwl/inputs/1290633107/proteome.bed --refdir /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/neoscan/C3L-00677/cromwell-workdir/cromwell-executions/neoscan.cwl/6994fb4a-0450-4dbb-818a-bcfdc8d73721/call-neoscan.cwl/inputs/881998829/refseq_hg38_june29 --optitype /miniconda/envs/neoscan/bin/OptiPathPipeline.py --fallele /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/neoscan/C3L-00677/cromwell-workdir/cromwell-executions/neoscan.cwl/6994fb4a-0450-4dbb-818a-bcfdc8d73721/call-neoscan.cwl/inputs/737207470/allelelist --netmhc /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/neoscan/C3L-00677/cromwell-workdir/cromwell-executions/neoscan.cwl/6994fb4a-0450-4dbb-818a-bcfdc8d73721/call-neoscan.cwl/inputs/-214520106/netMHC --fopticonfig /pecgs-neoscan/src/neoscan/config.ini --bam 1 --rna 0 --log /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/neoscan/C3L-00677/cromwell-workdir/cromwell-executions/neoscan.cwl/6994fb4a-0450-4dbb-818a-bcfdc8d73721/call-neoscan.cwl/execution/logs --step 4\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
