{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e0d15dd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "ca9d1196",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "e37986f7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomAD_AF</th>\n",
       "      <th>gnomAD_AFR_AF</th>\n",
       "      <th>gnomAD_AMR_AF</th>\n",
       "      <th>gnomAD_ASJ_AF</th>\n",
       "      <th>gnomAD_EAS_AF</th>\n",
       "      <th>gnomAD_FIN_AF</th>\n",
       "      <th>gnomAD_NFE_AF</th>\n",
       "      <th>gnomAD_OTH_AF</th>\n",
       "      <th>gnomAD_SAS_AF</th>\n",
       "      <th>vcf_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PERM1</td>\n",
       "      <td>84808</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>979296</td>\n",
       "      <td>979296</td>\n",
       "      <td>+</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>979296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>UBE2J2</td>\n",
       "      <td>118424</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1255365</td>\n",
       "      <td>1255365</td>\n",
       "      <td>+</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000088</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000109</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000106</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>1255365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SCNN1D</td>\n",
       "      <td>6339</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1287237</td>\n",
       "      <td>1287237</td>\n",
       "      <td>+</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>0.000048</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.000169</td>\n",
       "      <td>0.000066</td>\n",
       "      <td>1287237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PLCH2</td>\n",
       "      <td>9651</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2504461</td>\n",
       "      <td>2504461</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000223</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000063</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2504461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PRDM16</td>\n",
       "      <td>63976</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3404787</td>\n",
       "      <td>3404787</td>\n",
       "      <td>+</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3404787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1088</th>\n",
       "      <td>IDH3G</td>\n",
       "      <td>3421</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chrX</td>\n",
       "      <td>153787514</td>\n",
       "      <td>153787514</td>\n",
       "      <td>+</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>153787514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1089</th>\n",
       "      <td>HCFC1</td>\n",
       "      <td>3054</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chrX</td>\n",
       "      <td>153952907</td>\n",
       "      <td>153952907</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>153952907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1090</th>\n",
       "      <td>FLNA</td>\n",
       "      <td>2316</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154354918</td>\n",
       "      <td>154354918</td>\n",
       "      <td>+</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>154354918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1091</th>\n",
       "      <td>EMD</td>\n",
       "      <td>2010</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154380946</td>\n",
       "      <td>154380946</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>154380946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1092</th>\n",
       "      <td>FAM50A</td>\n",
       "      <td>9130</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chrX</td>\n",
       "      <td>154450428</td>\n",
       "      <td>154450428</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>154450428</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1093 rows × 133 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  Start_Position  \\\n",
       "0          PERM1           84808      .     GRCh38       chr1          979296   \n",
       "1         UBE2J2          118424      .     GRCh38       chr1         1255365   \n",
       "2         SCNN1D            6339      .     GRCh38       chr1         1287237   \n",
       "3          PLCH2            9651      .     GRCh38       chr1         2504461   \n",
       "4         PRDM16           63976      .     GRCh38       chr1         3404787   \n",
       "...          ...             ...    ...        ...        ...             ...   \n",
       "1088       IDH3G            3421      .     GRCh38       chrX       153787514   \n",
       "1089       HCFC1            3054      .     GRCh38       chrX       153952907   \n",
       "1090        FLNA            2316      .     GRCh38       chrX       154354918   \n",
       "1091         EMD            2010      .     GRCh38       chrX       154380946   \n",
       "1092      FAM50A            9130      .     GRCh38       chrX       154450428   \n",
       "\n",
       "      End_Position Strand Variant_Classification Variant_Type  ... gnomAD_AF  \\\n",
       "0           979296      +                 Silent          SNP  ...       NaN   \n",
       "1          1255365      +                 Silent          SNP  ...  0.000088   \n",
       "2          1287237      +                 Silent          SNP  ...  0.000025   \n",
       "3          2504461      +      Missense_Mutation          SNP  ...  0.000049   \n",
       "4          3404787      +                 Silent          SNP  ...       NaN   \n",
       "...            ...    ...                    ...          ...  ...       ...   \n",
       "1088     153787514      +                 Silent          SNP  ...  0.000011   \n",
       "1089     153952907      +      Missense_Mutation          SNP  ...  0.000008   \n",
       "1090     154354918      +                 Silent          SNP  ...  0.000017   \n",
       "1091     154380946      +      Missense_Mutation          SNP  ...       NaN   \n",
       "1092     154450428      +      Missense_Mutation          SNP  ...       NaN   \n",
       "\n",
       "     gnomAD_AFR_AF gnomAD_AMR_AF gnomAD_ASJ_AF  gnomAD_EAS_AF gnomAD_FIN_AF  \\\n",
       "0              NaN           NaN           NaN            NaN           NaN   \n",
       "1              NaN      0.000145           NaN       0.000109           NaN   \n",
       "2              NaN           NaN           NaN       0.000056      0.000048   \n",
       "3         0.000065           NaN           NaN       0.000223           NaN   \n",
       "4              NaN           NaN           NaN            NaN           NaN   \n",
       "...            ...           ...           ...            ...           ...   \n",
       "1088           NaN           NaN           NaN            NaN           NaN   \n",
       "1089           NaN           NaN           NaN            NaN           NaN   \n",
       "1090           NaN           NaN           NaN            NaN           NaN   \n",
       "1091           NaN           NaN           NaN            NaN           NaN   \n",
       "1092           NaN           NaN           NaN            NaN           NaN   \n",
       "\n",
       "     gnomAD_NFE_AF gnomAD_OTH_AF gnomAD_SAS_AF    vcf_pos  \n",
       "0              NaN           NaN           NaN     979296  \n",
       "1         0.000106           NaN      0.000098    1255365  \n",
       "2         0.000009      0.000169      0.000066    1287237  \n",
       "3         0.000063           NaN           NaN    2504461  \n",
       "4              NaN           NaN           NaN    3404787  \n",
       "...            ...           ...           ...        ...  \n",
       "1088      0.000025           NaN           NaN  153787514  \n",
       "1089      0.000019           NaN           NaN  153952907  \n",
       "1090      0.000037           NaN           NaN  154354918  \n",
       "1091           NaN           NaN           NaN  154380946  \n",
       "1092           NaN           NaN           NaN  154450428  \n",
       "\n",
       "[1093 rows x 133 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maf = pd.read_csv('/data/sandbox/test.maf', sep='\\t', header=1)\n",
    "maf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "8e987a1d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele1</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1088</th>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1089</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1090</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1091</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1092</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1093 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Reference_Allele Tumor_Seq_Allele1 Tumor_Seq_Allele2\n",
       "0                   C                 C                 T\n",
       "1                   G                 G                 A\n",
       "2                   C                 C                 T\n",
       "3                   C                 C                 T\n",
       "4                   C                 C                 T\n",
       "...               ...               ...               ...\n",
       "1088                C                 C                 T\n",
       "1089                G                 G                 A\n",
       "1090                G                 G                 A\n",
       "1091                G                 G                 A\n",
       "1092                G                 G                 A\n",
       "\n",
       "[1093 rows x 3 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maf[['Reference_Allele',\n",
    " 'Tumor_Seq_Allele1',\n",
    " 'Tumor_Seq_Allele2']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "6542f196",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Hugo_Symbol',\n",
       " 'Entrez_Gene_Id',\n",
       " 'Center',\n",
       " 'NCBI_Build',\n",
       " 'Chromosome',\n",
       " 'Start_Position',\n",
       " 'End_Position',\n",
       " 'Strand',\n",
       " 'Variant_Classification',\n",
       " 'Variant_Type',\n",
       " 'Reference_Allele',\n",
       " 'Tumor_Seq_Allele1',\n",
       " 'Tumor_Seq_Allele2',\n",
       " 'dbSNP_RS',\n",
       " 'dbSNP_Val_Status',\n",
       " 'Tumor_Sample_Barcode',\n",
       " 'Matched_Norm_Sample_Barcode',\n",
       " 'Match_Norm_Seq_Allele1',\n",
       " 'Match_Norm_Seq_Allele2',\n",
       " 'Tumor_Validation_Allele1',\n",
       " 'Tumor_Validation_Allele2',\n",
       " 'Match_Norm_Validation_Allele1',\n",
       " 'Match_Norm_Validation_Allele2',\n",
       " 'Verification_Status',\n",
       " 'Validation_Status',\n",
       " 'Mutation_Status',\n",
       " 'Sequencing_Phase',\n",
       " 'Sequence_Source',\n",
       " 'Validation_Method',\n",
       " 'Score',\n",
       " 'BAM_File',\n",
       " 'Sequencer',\n",
       " 'Tumor_Sample_UUID',\n",
       " 'Matched_Norm_Sample_UUID',\n",
       " 'HGVSc',\n",
       " 'HGVSp',\n",
       " 'HGVSp_Short',\n",
       " 'Transcript_ID',\n",
       " 'Exon_Number',\n",
       " 't_depth',\n",
       " 't_ref_count',\n",
       " 't_alt_count',\n",
       " 'n_depth',\n",
       " 'n_ref_count',\n",
       " 'n_alt_count',\n",
       " 'all_effects',\n",
       " 'Allele',\n",
       " 'Gene',\n",
       " 'Feature',\n",
       " 'Feature_type',\n",
       " 'Consequence',\n",
       " 'cDNA_position',\n",
       " 'CDS_position',\n",
       " 'Protein_position',\n",
       " 'Amino_acids',\n",
       " 'Codons',\n",
       " 'Existing_variation',\n",
       " 'ALLELE_NUM',\n",
       " 'DISTANCE',\n",
       " 'STRAND_VEP',\n",
       " 'SYMBOL',\n",
       " 'SYMBOL_SOURCE',\n",
       " 'HGNC_ID',\n",
       " 'BIOTYPE',\n",
       " 'CANONICAL',\n",
       " 'CCDS',\n",
       " 'ENSP',\n",
       " 'SWISSPROT',\n",
       " 'TREMBL',\n",
       " 'UNIPARC',\n",
       " 'RefSeq',\n",
       " 'SIFT',\n",
       " 'PolyPhen',\n",
       " 'EXON',\n",
       " 'INTRON',\n",
       " 'DOMAINS',\n",
       " 'AF',\n",
       " 'AFR_AF',\n",
       " 'AMR_AF',\n",
       " 'ASN_AF',\n",
       " 'EAS_AF',\n",
       " 'EUR_AF',\n",
       " 'SAS_AF',\n",
       " 'AA_AF',\n",
       " 'EA_AF',\n",
       " 'CLIN_SIG',\n",
       " 'SOMATIC',\n",
       " 'PUBMED',\n",
       " 'MOTIF_NAME',\n",
       " 'MOTIF_POS',\n",
       " 'HIGH_INF_POS',\n",
       " 'MOTIF_SCORE_CHANGE',\n",
       " 'IMPACT',\n",
       " 'PICK',\n",
       " 'VARIANT_CLASS',\n",
       " 'TSL',\n",
       " 'HGVS_OFFSET',\n",
       " 'PHENO',\n",
       " 'MINIMISED',\n",
       " 'ExAC_AF',\n",
       " 'ExAC_AF_AFR',\n",
       " 'ExAC_AF_AMR',\n",
       " 'ExAC_AF_EAS',\n",
       " 'ExAC_AF_FIN',\n",
       " 'ExAC_AF_NFE',\n",
       " 'ExAC_AF_OTH',\n",
       " 'ExAC_AF_SAS',\n",
       " 'GENE_PHENO',\n",
       " 'FILTER',\n",
       " 'flanking_bps',\n",
       " 'vcf_id',\n",
       " 'vcf_qual',\n",
       " 'ExAC_AF_Adj',\n",
       " 'ExAC_AC_AN_Adj',\n",
       " 'ExAC_AC_AN',\n",
       " 'ExAC_AC_AN_AFR',\n",
       " 'ExAC_AC_AN_AMR',\n",
       " 'ExAC_AC_AN_EAS',\n",
       " 'ExAC_AC_AN_FIN',\n",
       " 'ExAC_AC_AN_NFE',\n",
       " 'ExAC_AC_AN_OTH',\n",
       " 'ExAC_AC_AN_SAS',\n",
       " 'ExAC_FILTER',\n",
       " 'gnomAD_AF',\n",
       " 'gnomAD_AFR_AF',\n",
       " 'gnomAD_AMR_AF',\n",
       " 'gnomAD_ASJ_AF',\n",
       " 'gnomAD_EAS_AF',\n",
       " 'gnomAD_FIN_AF',\n",
       " 'gnomAD_NFE_AF',\n",
       " 'gnomAD_OTH_AF',\n",
       " 'gnomAD_SAS_AF',\n",
       " 'vcf_pos']"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(maf.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "95106917",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'DEL', 'DNP', 'INS', 'SNP'}"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(maf['Variant_Type'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e1b2e60",
   "metadata": {},
   "outputs": [],
   "source": [
    "# chromosome, start position, ref allele, alt allele, gene hugo symbol, HGSV short, is it somatic or germline mutation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "4824473b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>979296</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>p.V578=</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1255365</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UBE2J2</td>\n",
       "      <td>p.N222=</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1287237</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SCNN1D</td>\n",
       "      <td>p.H416=</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2504461</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PLCH2</td>\n",
       "      <td>p.R1167C</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3404787</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>p.C311=</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1088</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153787514</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>IDH3G</td>\n",
       "      <td>p.A208=</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1089</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153952907</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>HCFC1</td>\n",
       "      <td>p.R1517W</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1090</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154354918</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>FLNA</td>\n",
       "      <td>p.F1708=</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1091</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154380946</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>EMD</td>\n",
       "      <td>p.A172T</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1092</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154450428</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>FAM50A</td>\n",
       "      <td>p.R339H</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1003 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chromosome  Start_Position Reference_Allele Tumor_Seq_Allele2  \\\n",
       "0          chr1          979296                C                 T   \n",
       "1          chr1         1255365                G                 A   \n",
       "2          chr1         1287237                C                 T   \n",
       "3          chr1         2504461                C                 T   \n",
       "4          chr1         3404787                C                 T   \n",
       "...         ...             ...              ...               ...   \n",
       "1088       chrX       153787514                C                 T   \n",
       "1089       chrX       153952907                G                 A   \n",
       "1090       chrX       154354918                G                 A   \n",
       "1091       chrX       154380946                G                 A   \n",
       "1092       chrX       154450428                G                 A   \n",
       "\n",
       "     Hugo_Symbol HGVSp_Short     type  \n",
       "0          PERM1     p.V578=  Somatic  \n",
       "1         UBE2J2     p.N222=  Somatic  \n",
       "2         SCNN1D     p.H416=  Somatic  \n",
       "3          PLCH2    p.R1167C  Somatic  \n",
       "4         PRDM16     p.C311=  Somatic  \n",
       "...          ...         ...      ...  \n",
       "1088       IDH3G     p.A208=  Somatic  \n",
       "1089       HCFC1    p.R1517W  Somatic  \n",
       "1090        FLNA    p.F1708=  Somatic  \n",
       "1091         EMD     p.A172T  Somatic  \n",
       "1092      FAM50A     p.R339H  Somatic  \n",
       "\n",
       "[1003 rows x 7 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "valid = ['SNP']\n",
    "snp = maf[[True if s in valid else False\n",
    "          for s in maf['Variant_Type']]]\n",
    "snp = snp[['Chromosome', 'Start_Position', 'Reference_Allele', 'Tumor_Seq_Allele2', 'Hugo_Symbol', 'HGVSp_Short']]\n",
    "snp['type'] = 'Somatic'\n",
    "snp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "a907724a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>15946185</td>\n",
       "      <td>CTC</td>\n",
       "      <td>-</td>\n",
       "      <td>ZBTB17</td>\n",
       "      <td>p.E168del</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26773685</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>p.P1326Rfs*155</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>43426503</td>\n",
       "      <td>TGAG</td>\n",
       "      <td>-</td>\n",
       "      <td>SZT2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>61332581</td>\n",
       "      <td>CA</td>\n",
       "      <td>-</td>\n",
       "      <td>NFIA</td>\n",
       "      <td>p.Q278Dfs*51</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>84866081</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>LPAR3</td>\n",
       "      <td>p.Y14Ifs*18</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1055</th>\n",
       "      <td>chrX</td>\n",
       "      <td>70426380</td>\n",
       "      <td>CTC</td>\n",
       "      <td>-</td>\n",
       "      <td>GDPD2</td>\n",
       "      <td>p.L126del</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1060</th>\n",
       "      <td>chrX</td>\n",
       "      <td>80030540</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>TBX22</td>\n",
       "      <td>p.S333Lfs*3</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1062</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85270708</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>ZNF711</td>\n",
       "      <td>p.F438Vfs*2</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1065</th>\n",
       "      <td>chrX</td>\n",
       "      <td>102716543</td>\n",
       "      <td>TCCT</td>\n",
       "      <td>-</td>\n",
       "      <td>GPRASP2</td>\n",
       "      <td>p.S560Tfs*31</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1082</th>\n",
       "      <td>chrX</td>\n",
       "      <td>136244784</td>\n",
       "      <td>TTTC</td>\n",
       "      <td>-</td>\n",
       "      <td>MAP7D3</td>\n",
       "      <td>p.E88Hfs*25</td>\n",
       "      <td>Somatic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>89 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chromosome  Start_Position Reference_Allele Tumor_Seq_Allele2  \\\n",
       "17         chr1        15946185              CTC                 -   \n",
       "25         chr1        26773685                C                 -   \n",
       "32         chr1        43426503             TGAG                 -   \n",
       "35         chr1        61332581               CA                 -   \n",
       "36         chr1        84866081                A                 -   \n",
       "...         ...             ...              ...               ...   \n",
       "1055       chrX        70426380              CTC                 -   \n",
       "1060       chrX        80030540                C                 -   \n",
       "1062       chrX        85270708                -                 A   \n",
       "1065       chrX       102716543             TCCT                 -   \n",
       "1082       chrX       136244784             TTTC                 -   \n",
       "\n",
       "     Hugo_Symbol     HGVSp_Short     type  \n",
       "17        ZBTB17       p.E168del  Somatic  \n",
       "25        ARID1A  p.P1326Rfs*155  Somatic  \n",
       "32          SZT2             NaN  Somatic  \n",
       "35          NFIA    p.Q278Dfs*51  Somatic  \n",
       "36         LPAR3     p.Y14Ifs*18  Somatic  \n",
       "...          ...             ...      ...  \n",
       "1055       GDPD2       p.L126del  Somatic  \n",
       "1060       TBX22     p.S333Lfs*3  Somatic  \n",
       "1062      ZNF711     p.F438Vfs*2  Somatic  \n",
       "1065     GPRASP2    p.S560Tfs*31  Somatic  \n",
       "1082      MAP7D3     p.E88Hfs*25  Somatic  \n",
       "\n",
       "[89 rows x 7 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "valid = ['INS', 'DEL']\n",
    "indel = maf[[True if s in valid else False\n",
    "          for s in maf['Variant_Type']]]\n",
    "indel = indel[['Chromosome', 'Start_Position', 'Reference_Allele', 'Tumor_Seq_Allele2', 'Hugo_Symbol', 'HGVSp_Short']]\n",
    "indel['type'] = 'Somatic'\n",
    "indel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "856f75a3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "442e940a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa141989",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d02f538f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4041317b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "493cef22",
   "metadata": {},
   "outputs": [],
   "source": [
    "import vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "efcc258c",
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf_reader = vcf.Reader(open('/data/sandbox/test_vcf.vcf'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "959753f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# chromosome, start position, ref allele, alt allele, gene hugo symbol, HGSV short, is it somatic or germline mutation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "200f7ce9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Record(CHROM=chr1, POS=979296, REF=C, ALT=[T])\n"
     ]
    }
   ],
   "source": [
    "data = []\n",
    "r = None\n",
    "for record in vcf_reader:\n",
    "    r = record\n",
    "    print(record)\n",
    "    break\n",
    "    data.append([record.CHROM, record.POS, record.REF, record.ALT, ])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "479a1a7d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r.FILTER"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f50ecce4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'GT:AD:DP:DP4:FREQ'"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r.FORMAT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "7678533a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Call(sample=TUMOR, CallData(GT=0/1, AD=[49, 6], DP=55, DP4=2,47,0,6, FREQ=10.91%))]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r.get_hets()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d8912e54",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'AC': [1],\n",
       " 'AF': [0.25],\n",
       " 'AN': 4,\n",
       " 'DP': 390,\n",
       " 'SOMATIC': True,\n",
       " 'set': 'varscan-mutect-strelka',\n",
       " 'CSQ': ['T|synonymous_variant|LOW|PERM1|ENSG00000187642|Transcript|ENST00000341290|protein_coding|3/5||ENST00000341290.6:c.1392G>A|ENSP00000343864.2:p.Val464=|1428/3035|1392/2031|464/676|V|gtG/gtA||1||-1|||SNV|1|HGNC|HGNC:28208||2|A2||Q5SV97||UPI000022DAF4|NM_001291367.2||||||||||||||||||||||||||||||||||||||||||||||||',\n",
       "  'T|downstream_gene_variant|MODIFIER|PLEKHN1|ENSG00000187583|Transcript|ENST00000379407|protein_coding|||||||||||1|4288|1|||SNV|1|HGNC|HGNC:25284||1|A2|CCDS53256.1|Q494U1||UPI00005764FF|NM_001160184.2||||||||||||||||||||||||||||||||||||||||||||||||',\n",
       "  'T|downstream_gene_variant|MODIFIER|PLEKHN1|ENSG00000187583|Transcript|ENST00000379409|protein_coding|||||||||||1|4288|1|||SNV|1|HGNC|HGNC:25284||2|A2||Q494U1||UPI0000D61E06|||||||||||||||||||||||||||||||||||||||||||||||||',\n",
       "  'T|downstream_gene_variant|MODIFIER|PLEKHN1|ENSG00000187583|Transcript|ENST00000379410|protein_coding|||||||||||1|3431|1|||SNV|1|HGNC|HGNC:25284|YES|1|P3|CCDS4.1|Q494U1||UPI00001416D8|NM_001367552.1&NM_032129.3||||||||||||||||||||||||||||||||||||||||||||||||',\n",
       "  'T|synonymous_variant|LOW|PERM1|ENSG00000187642|Transcript|ENST00000433179|protein_coding|1/3||ENST00000433179.3:c.1734G>A|ENSP00000414022.3:p.Val578=|1734/3340|1734/2373|578/790|V|gtG/gtA||1||-1||1|SNV|1|HGNC|HGNC:28208|YES|5|P2|CCDS76083.1|Q5SV97||UPI0003E30FA7|NM_001291366.2&NM_001369897.1&NM_001369898.1||||||||||||||||||||||||||||||||||||||||||||||||',\n",
       "  'T|upstream_gene_variant|MODIFIER|PERM1|ENSG00000187642|Transcript|ENST00000479361|retained_intron|||||||||||1|2655|-1|||SNV|1|HGNC|HGNC:28208||1||||||||||||||||||||||||||||||||||||||||||||||||||||||',\n",
       "  'T|downstream_gene_variant|MODIFIER|PLEKHN1|ENSG00000187583|Transcript|ENST00000491024|protein_coding|||||||||||1|3431|1|cds_start_NF||SNV|1|HGNC|HGNC:25284||3||||J3KSM5|UPI000268AE1F|||||||||||||||||||||||||||||||||||||||||||||||||']}"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r.INFO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "8a4d59c0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r.is_snp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b7ee751c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r.is_indel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "71ee1f2e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ts'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r.var_subtype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "d83ff0d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pyensembl install --release 102 --species human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "ebd0f1e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pyensembl import EnsemblRelease\n",
    "\n",
    "data = EnsemblRelease(102, )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "fe0859fa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('chr1', 979296)"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r.CHROM, r.POS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "ceebcb54",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_names = data.gene_names_at_locus(contig=r.CHROM.replace('chr', ''), position=r.POS, )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "bbd38998",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['PERM1']"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "5405b52a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Gene(gene_id='ENSG00000187642', gene_name='PERM1', biotype='protein_coding', contig='1', start=975204, end=982093, strand='-', genome='GRCh38')]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.genes_at_locus(contig=r.CHROM.replace('chr', ''), position=r.POS, )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "9ac5f0d5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['HLA-A']"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.gene_names_at_locus(contig=6, position=29945884)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ec4d255c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
